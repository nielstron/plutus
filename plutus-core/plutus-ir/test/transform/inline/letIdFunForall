<<<<<<< HEAD
{- 
A single occurrence polymorphic identity function. It should get inlined by the unconditional 
inlining optimization.

let
    idFun :: forall a. a -> a
    idFun x = x
in idFun @Integer 3
-}

=======
>>>>>>> aa7efa877 (Add id fully applied test.)
(let
    (nonrec)
    (termbind
      (strict)
      (vardecl idFun (all a (type) (fun a a)))
      (abs a (type) (lam x a x))
    )
<<<<<<< HEAD
    [ { idFun (con integer) } (con integer 3) ]
=======
    [ { idFun (fun (con integer) (con integer)) } (lam x (con integer) x) ]
>>>>>>> aa7efa877 (Add id fully applied test.)
)