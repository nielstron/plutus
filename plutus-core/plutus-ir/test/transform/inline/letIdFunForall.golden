<<<<<<< HEAD
[ { (abs a (type) (lam x a x)) (con integer) } (con integer 3) ]
=======
[
  { (abs a (type) (lam x a x)) (fun (con integer) (con integer)) }
  (lam x (con integer) x)
]
>>>>>>> aa7efa877 (Add id fully applied test.)
